.gitlab-ci.yml

```diff
@@ -35,98 +33,78 @@
       echo "Finished job with code $res"
       exit $res
 
+# CI constraints for every branch
+workflow:
+  rules:
+      # Only run if source is an open merge request or a branch push
+    - if: $CI_MERGE_REQUEST_IID
+    - if: $CI_OPEN_MERGE_REQUESTS
+      when: never
+    - if: $CI_COMMIT_BRANCH
```

Comment:
Workflow reules are unnecessary here since you trigger the pipelines manually, and push with the skip-ci option in the push mirror

Reply:
More to the point, GitLab has no knowledge of the state of GitHub merge requests, so the rules are meaningless in this case.